client/jetpack-connect/store-header.tsx

```diff
@@ -23,6 +28,15 @@ export default function StoreHeader(): React.ReactElement {
 		'add-bottom-margin': ! isStoreLanding,
 	} );
 
+	const useV2Banner = config.isEnabled( 'jetpack/pricing-page-v2-banner' );
+
+	const renderBanner = () => {
+		if ( useV2Banner ) {
+			return <IntroPricingBannerV2 jetpackSaleCoupon={ jetpackSaleCoupon } />;
+		}
+		return jetpackSaleCoupon !== null ? null : <IntroPricingBanner />;
+	};
```

Comment:
`renderBanner` will be recreated on every render and thus it will trigger more re-renders. It is better to extract it into its own small component: like this

```tsx
const RenderBanner: React.FC = () => {
	const jetpackSaleCoupon = useSelector( getJetpackSaleCoupon );
	const useV2Banner = config.isEnabled( 'jetpack/pricing-page-v2-banner' );

	if ( useV2Banner ) {
		return <IntroPricingBannerV2 jetpackSaleCoupon={ jetpackSaleCoupon } />;
	}
	return jetpackSaleCoupon !== null ? null : <IntroPricingBanner />;
};
```

Then use it below as `<RenderBanner />`.

Reply:
good catch! Another alternative would be to pull out the original arrow function outside of React Component and just pass in the need variables as arguments. However, I ended up moving the `<IntroPricingBannerV2 />` to another place, removing the need for this logic.